10.1/root/usr/share/container-scripts/mysql/check-upgrade.sh

```diff
@@ -0,0 +1,60 @@
+check_datadir_version() {
+  local datadir="$1"
+  local datadir_version=$(get_datadir_version "$datadir")
+  local mysqld_version=$(mysqld_compat_version)
+
+  for upgrade_action in ${MYSQL_UPGRADE//,/ } ; do
+    case ${upgrade_action} in
+      auto|warn)
+        if [ -z "${datadir_version}" ] || [ "${datadir_version}" -eq 0 ] ; then
+          echo "Warning: Version of the data could not be determined. Running such a container is very risky. The current daemon version is ${mysqld_version}. If you are sure that the data dir format corresponds with this version, restore the data from a back-up. If restoring from a back-up is not possible, create a file 'mysql_upgrade_info' that includes version information ('${mysqld_version}' in this case) in the root of the data directory. To do so, run 'mysql_upgrade' utility that checks the tables and creates such a file as well. That will enable correct upgrade check in the future."
+          continue
+        fi
+
+        if [ "${datadir_version}" -eq "${mysqld_version}" ] ; then
+          log_info 'MySQL server version check OK.'
+          continue
+        fi
+
+        if [ $(( ${datadir_version} + 1 )) -eq "${mysqld_version}" -o "${datadir_version}" -eq 505 -a "${mysqld_version}" -eq 1000 ] ; then
+          log_info "MySQL server is version '${mysqld_version}' and datadir is version '${datadir_version}'."
```

Comment:
this doesn't strike me as being obviously a warning.  it seems pretty easy to overlook/ignore.